C# plugin improvements

cklam2 opened the pull request:
Solves #4214. 

Additionally added some more improvements:
- Nullable value types
- Initial value support
- Class and field documentation in summary block
- Deprecated fields have DeprecatedAttribute
- Required input fields have JsonRequiredAttribute
- Generic implementation to compose input object
- Code in namespace indented and wrapped

There's one incompatibility though: the generated C# method `getInputObject` is renamed to `GetInputObject` just to follow the general C# naming convention. If breaking compatibility is a no-go to you, it can be reverted.

dotansimha commented:
This is great! Thank you @cklam2 ! 
Any chance you can add some unit tests to make sure we are not breaking these changes in the future?

cklam2 commented:
@dotansimha
Yep, already added some unit tests (locally) for the fixes but they are not complete yet. Eg missing tests for existing code. Ok to do another pull request once they are done? Or you want it be included in this one?

dotansimha commented:
I think if tests are missing for the existing code, we can improve it in a future PR. Let me know when you push the tests for the modified code, and we'll progress on that.

Thank you!

cklam2 commented:
Well, glad that I extended the unit tests cause found some regressions in the initial values part and the array composition. Also refactored some of the code, hence all those changes.

Also committed the unit test for the c-sharp plugin. Works locally. Only not sure if I need to commit a `jest.config.js` or need to change `package.json`. @dotansimha can you check if the unit test needs some configuration in the project? 

Rebased the branch to latest upstream master.

dotansimha commented:
This is great, thank you so much @cklam2 !
